lib/riemann/helpers/event.ex

```diff
@@ -17,28 +17,9 @@ defmodule Riemann.Helpers.Event do
       end
 
       def build(dict) do
-        # i'd move this to a module attribute, but we can't assume that the build host is the runtime host.
-        # could also stick it in the process dictionary if it's an issue
-        {:ok, hostname} = :inet.gethostname
-        hostname = @event_host || :erlang.list_to_binary(hostname)
-
-        dict = Enum.into(dict, %{})
-        dict = Map.merge(%{host: hostname, time: :erlang.system_time(:seconds)}, dict)
-        dict = case Map.get(dict, :attributes) do
-                 nil -> dict
-                 a   -> Map.put(dict, :attributes, Attribute.build(a))
-               end
-
-        case Map.get(dict, :metric) do
-          i when is_integer(i) -> Map.put(dict, :metric_sint64, i)
-          f when is_float(f)   -> Map.put(dict, :metric_d, f)
-          nil -> raise ArgumentError, "no metric provided for dict #{inspect dict}"
-        end
-        |> Map.to_list
-        |> new()
+        Riemann.Helpers.Event.build(dict, __MODULE__)
```

Comment:
That makes sense for maintainability. Tricky that __MODULE__ means two different things on the same line. I'll make sure to add a comment.